src/pk/asn1/oid/pk_get_oid.c

```diff
@@ -0,0 +1,45 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+#include "tomcrypt_private.h"
+
+#ifdef LTC_DER
+
+typedef struct {
+   int pka;
+   const char* oid;
+} pka_oid;
+
+static const pka_oid oids[] = {
+                               { PKA_RSA, "1.2.840.113549.1.1.1" },
+                               { PKA_DSA, "1.2.840.10040.4.1" },
+                               { PKA_EC, "1.2.840.10045.2.1" },
+                               { PKA_EC_PRIMEF, "1.2.840.10045.1.1" },
+};
+
+/*
+   Returns the OID of the public key algorithm.
```

Comment:
cosmetics: just use something more general than "public key algorithm" as especially in the new PKCS#8 implementation we will used for OIDs of many other kinds.

Reply:
very true

Reply:
better? fb7b8799cd0258741307eb8c89cfd85deab5ba5b